Dockerfile

```diff
@@ -1,86 +1,381 @@
+ENV GOENV_DISABLE_GOPATH=1
+ENV GOPATH="/go"
+
+#=======================End of layer: tools  =================
+FROM tools AS runtimes
+
+#****************     .NET-CORE     *******************************************************
+
+ENV DOTNET_6_SDK_VERSION="6.0.405"
+ENV DOTNET_ROOT="/root/.dotnet"
+
+# Add .NET Core 6 Global Tools install folder to PATH
+RUN  /usr/local/bin/dotnet-install.sh -v $DOTNET_6_SDK_VERSION \
+     && dotnet --list-sdks \
+     && rm -rf /tmp/*
+
+# Trigger the population of the local package cache
+ENV NUGET_XMLDOC_MODE skip
+RUN set -ex \
+    && mkdir warmup \
+    && cd warmup \
+    && dotnet new \
+    && cd .. \
+    && rm -rf warmup \
+    && rm -rf /tmp/NuGetScratch
+
+# Install Powershell Core
+# See instructions at https://docs.microsoft.com/en-us/powershell/scripting/setup/installing-powershell-core-on-linux
+ARG POWERSHELL_VERSION=7.2.8
+ARG POWERSHELL_DOWNLOAD_URL=https://github.com/PowerShell/PowerShell/releases/download/v$POWERSHELL_VERSION/powershell-$POWERSHELL_VERSION-linux-x64.tar.gz
+ARG POWERSHELL_DOWNLOAD_SHA=28FF2653667AC63B508F0B98433E48F64E6BC1EC59F8C1D252BA89EB5A7441A2
+
+RUN set -ex \
+    && curl -SL $POWERSHELL_DOWNLOAD_URL --output powershell.tar.gz \
+    && echo "$POWERSHELL_DOWNLOAD_SHA powershell.tar.gz" | sha256sum -c - \
+    && mkdir -p /opt/microsoft/powershell/$POWERSHELL_VERSION \
+    && tar zxf powershell.tar.gz -C /opt/microsoft/powershell/$POWERSHELL_VERSION \
+    && rm powershell.tar.gz \
+    && ln -s /opt/microsoft/powershell/$POWERSHELL_VERSION/pwsh /usr/bin/pwsh
+#****************     END .NET-CORE     *******************************************************
+
+
+#****************      NODEJS     ****************************************************
+
+ENV NODE_18_VERSION="18.15.0"
+
+RUN  n $NODE_18_VERSION && npm install --save-dev -g -f grunt && npm install --save-dev -g -f grunt-cli && npm install --save-dev -g -f webpack \
+     && curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
+     && echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list \
+     && apt-get update && apt-get install -y -qq --no-install-recommends yarn \
+     && yarn --version \
+     && cd / && rm -rf $N_SRC_DIR && rm -rf /tmp/*
+
+#****************      END NODEJS     ****************************************************
+
+#**************** RUBY *********************************************************
+
+ENV RUBY_31_VERSION="3.1.3"
+
+RUN rbenv install $RUBY_31_VERSION && rm -rf /tmp/* \
+    && rbenv global $RUBY_31_VERSION && ruby -v
+
+#**************** END RUBY *****************************************************
+
+#**************** PYTHON *****************************************************
+# inspired by: https://github.com/aws/aws-codebuild-docker-images/blob/9282872af78aeb1b5df3010ed3872c40f3d0f056/al2/x86_64/standard/2.0/Dockerfile
+ENV PYTHON_38_VERSION="3.8.10"
```

Comment:
Python 3.8 is a bit older - so I had to dig into the older CodeBuild images to figure out how to configure this.
Here is a link to the dockerfile from Ubuntu standard 2.0, which used python 3.8:
https://github.com/aws/aws-codebuild-docker-images/blob/9282872af78aeb1b5df3010ed3872c40f3d0f056/al2/x86_64/standard/2.0/Dockerfile